cmake/EthCompilerSettings.cmake

```
@@ -192,11 +192,37 @@ endif ()
 
 if (SANITIZE)
 	# Perform case-insensitive string compare
-	string(TOLOWER "${SANITIZE}" san)
+	string(TOLOWER "${SANITIZE}" sanitizer)
 	# -fno-omit-frame-pointer gives more informative stack trace in case of an error
 	# -fsanitize-address-use-after-scope throws an error when a variable is used beyond its scope
-	if (san STREQUAL "address")
+	if (sanitizer STREQUAL "address")
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-omit-frame-pointer -fsanitize=address -fsanitize-address-use-after-scope")
+	elseif (sanitizer STREQUAL "undefined")
+		list(APPEND undefinedSanitizerFlags
```

Just a tiny nitpick: "flags" makes me think of a set of parameters, like the whole `-fsanitize=... -fsanitize-address-use-after-scope` part.

[UndefinedBehaviorSanitizer in clang docs](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html) refers to these sanitizer options as "checks" instead.
```suggestion
		list(APPEND undefinedSanitizerChecks
```